frame/aura/src/lib.rs

```diff
@@ -165,7 +187,7 @@ impl<T: Config> sp_runtime::BoundToRuntimeAppPublic for Pallet<T> {
 	type Public = T::AuthorityId;
 }
 
-impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
+impl<T: Config> OneSessionHandler<T::AccountId, T::MaxAuthorities> for Pallet<T> {
```

Comment:
I think this is very very good because now `OneSessionHandler<_, X>` is a different trait than `OneSessionHandler<_, Y>` and we should get some good compile time checks, ensuring that we can't faff with them: 

if babe implements `OneSessionHandler<_, 10>` and aura implements `OneSessionHandler<_, 20>`, I think if you then say `impl session::Config { type SessionHandlers = (babe, aura) }` we should get an error 🤔

Reply:
But the missing point is: 

- If session's `MaxAuthority` is 100
- and babe and aura's `MaxAuthority` is 50

I think we should prevent this at compile time but I don't it in the current state of the PR.

Reply:
oh no we have this guarantee as well, because of 

```
type SessionHandler: SessionHandler<Self::ValidatorId, Self::MaxValidators>;
```

Reply:
Yes, the point here is that there should be a single global config of `MaxAuthorities` / `MaxValidators` in the runtime, and this should be shared with all of the pallets which are using a session handler.

Anything else would result in a compile error, which ensures that these systems are working together correctly.

I dont think it would ever be that session and babe would have different max authorities, but maybe @andresilva could give insight

Reply:
Yeah, we should have an uniform number of authorities throughout the whole runtime.